src/api.c

```diff
@@ -469,7 +470,17 @@ static struct nmd_frame *ret_frame(struct nmd_ctx *s, AVFrame *frame, int status
     struct nmd_frame *ret = NULL;
     const struct nmdi_opts *o = &s->opts;
 
+    if (status == 0)
+        s->loop = 0;
+
     if (!frame) {
+        /*
+         * If the async module reached EOF and there is no more frame to
+         * return, we need to seek back to the beginning of the file because
+         * nmd_get_next_frame() is expected to loop around when reaching EOF.
+         */
+        if (status == AVERROR_EOF || status == AVERROR_EXIT)
+            s->loop = 1;
```

Comment:
Why not simply set `s->eof = !frame && (status == AVERROR_EOF || status == AVERROR_EXIT)` unconditionally?

Reply:
Done.